sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileIO.java

```diff
@@ -305,19 +306,21 @@ public MatchAll continuously(
 
     @Override
     public PCollection<MatchResult.Metadata> expand(PCollection<String> input) {
+      PCollection<MatchResult.Metadata> res;
       if (getConfiguration().getWatchInterval() == null) {
-        return input.apply(
+        res = input.apply(
             "Match filepatterns",
             ParDo.of(new MatchFn(getConfiguration().getEmptyMatchTreatment())));
       } else {
-        return input
+        res = input
             .apply(
                 "Continuously match filepatterns",
                 Watch.growthOf(new MatchPollFn())
                     .withPollInterval(getConfiguration().getWatchInterval())
                     .withTerminationPerInput(getConfiguration().getWatchTerminationCondition()))
             .apply(Values.<MatchResult.Metadata>create());
       }
+      return res.apply(Reshuffle.<MatchResult.Metadata>viaRandomKey());
```

Comment:
This introduces a reshuffle where we previously didn't do one. Would probably have been better as part of a separate PR since this changes behavior...

Reply:
Not sure it's worth extracting in a separate PR. Adding a fusion break to match was the main point of this PR and then I realized I could also deduplicate this between the many places.

Reply:
Is Match used in Streaming pipelines? If yes, adding it is update incompatible.

Reply:
Match did not exist in Beam 2.1